test/spec/app_spec.rb

```diff
@@ -830,6 +788,150 @@
         expect(elems.size).to be > 1
       }
+    end    
+
+    it "should redirect to team page if user asking to join a team is already a member", bin4:true do
+      page = Page.new(config: @config, driver: @driver)
+      page.go(@config['api_path'] + '/test/session?email=new'+@user_mail)
+      page = TeamsPage.new(config: @config, driver: @driver).load
+      @driver.navigate.to @config['self_url'] + "/"+@team1_slug+"/join"
+      sleep 3
+      @wait.until {
+        expect(@driver.current_url.eql? @config['self_url']+"/"+@team1_slug ).to be(true)
+      }    
+    end
+
+    it "should reject member to join team", bin4:true do
+      user = api_register_and_login_with_email
+      page = TeamsPage.new(config: @config, driver: @driver).load
+      page.ask_join_team(subdomain: @team1_slug)
+      @wait.until {
+        expect(@driver.find_element(:class, "message").nil?).to be(false)
+      }
+      api_logout
+      @driver.quit
+      @driver = new_driver(webdriver_url,browser_capabilities)
+      page = Page.new(config: @config, driver: @driver)
+      page.go(@config['api_path'] + '/test/session?email='+@user_mail)
+      #As the group creator, go to the members page and approve the joining request.
+      page = TeamsPage.new(config: @config, driver: @driver).load
+          .disapprove_join_team(subdomain: @team1_slug)
+      @wait.until {
+        expect(@driver.page_source.include?('Requests to join')).to be(false)
+      }      
+    end
+
+    #TO DO PROBLEMS WITH MENU
+    it "should change member role", bin3:true  do
+      utp = api_create_team
+      user2 = api_register_and_login_with_email
+      page = TeamsPage.new(config: @config, driver: @driver).load
+      page.ask_join_team(subdomain: utp[:team].slug)
+      sleep 3        
+      api_logout
+      @driver.quit
+
+      @driver = new_driver(webdriver_url,browser_capabilities)
+      page = Page.new(config: @config, driver: @driver)
+      page.go(@config['api_path'] + '/test/session?email='+utp[:user].email)
+      #As the group creator, go to the members page and approve the joining request.
+      page = TeamsPage.new(config: @config, driver: @driver).load
+      page.approve_join_team(subdomain: utp[:team].slug)      
+      sleep 3
+      expect(@driver.page_source.include?('Editor')).to be(false)
+      page.edit_user_role(subdomain: @team1_slug)
+      sleep 5
+      expect(@driver.page_source.include?('Editor')).to be(true)
+    end
+
+    it "should delete member from team", bin4:true do
+      page = Page.new(config: @config, driver: @driver)
+      page.go(@config['api_path'] + '/test/session?email='+@user_mail)
+      #As the group creator, go to the members page and approve the joining request.
+      sleep 1
+      page = TeamsPage.new(config: @config, driver: @driver).load
+          .delete_user_team(subdomain: @team1_slug)
+
+      expect(@driver.find_elements(:xpath => '//button').length == 4).to be(true)
```

Comment:
Would be good to add an `expect` before a user is rejected, to make sure that something changed.

Reply:
Both expects added to pages/team_page.rb